### PR TITLE
IDE-983 Fix Bookmark Logic

### PR DESCRIPTION
**Pull request by dehilsterlexis**

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith for code reivew. @JamesDeFabia I will get you an install to test the new bookmarks menus which should be "clearer" to the end user.

File: eclide/BookmarksView.h [hunk 1]
```diff
@@ -7,17 +7,24 @@
 #include <CustomMessages.h>
 
 const TCHAR * const CLEAR_BOOKMARKS_MSG = _T("Are you sure you want clear the displayed bookmarks?");
-const TCHAR * const SAVE_BOOKMARKS_MSG = _T("Are you sure you want overwrite the current bookmarks file?");
-const TCHAR * const LOAD_BOOKMARKS_MSG = _T("Are you sure you want replace the current list of bookmarks with what's saved?");
-const TCHAR * const LOAD_MERGE_BOOKMARKS_MSG = _T("Are you sure you want to merge your saved bookmarks with the current bookmark list?");
+const TCHAR * const SAVE_BOOKMARKS_OVERWRITE_MSG = _T("Are you sure you want overwrite the master bookmarks file?");
+const TCHAR * const SAVE_BOOKMARKS_MSG = _T("Are you sure you want to save the current booksmarks as a master bookmarks file?");
```
Comment (GordonSmith) [inline eclide/BookmarksView.h:11]: Typo "booksmarks"

File: eclide/BookmarksView.h [hunk 2]
```diff
@@ -103,9 +117,18 @@ class CBookmarksView:
     int GetSelectedCount();
     void Open();
     void Save(bool saveState=false);
-    void Load(bool loadState=false);
+    void Load(BM_FILE_TYPE fileType=BM_FILE_MASTER);
     void Clear();
     void OpenAttribute(int listRow);
+    bool BookmarkFilesState();
+    std::_tstring FileText(BM_FILE_TYPE fileType);
```
Comment (GordonSmith) [inline eclide/BookmarksView.h:124]: Change to:
```
const TCHAR * FileText(BM_FILE_TYPE fileType, const std::_tstring & result);
```

File: eclide/BookmarksView.cpp [hunk 3]
```diff
@@ -506,7 +513,7 @@ void CBookmarksView::ParseBookmarksEcl(const std::_tstring & ecl, const std::_ts
                         if (boost::algorithm::iequals(liner,nStr) && boost::algorithm::iequals(module,inModule) && boost::algorithm::iequals(attributeName,inAttributeName))
                         {
                             std::_tstring inComment = m_list.GetItemText(i, 6);
-                            if (boost::algorithm::iequals(comment,inComment))
+                            if (!boost::algorithm::iequals(comment,inComment))
```
Comment (GordonSmith) [inline eclide/BookmarksView.cpp:516]: Is this correct - not sure what is being checked?

File: eclide/ChildBuilderFrame.cpp [hunk 4]
```diff
@@ -1155,10 +1155,16 @@ bool RestoreExisting(IWorkspaceItem * workspaceItem, CChildBuilderFrm** pChild)
...
-                if (boost::algorithm::iequals(existingID, passedID))
+                bool existingIsFileFlag = boost::filesystem::exists(existingID) ? true : false;
+                if (existingIsFileFlag && boost::algorithm::iequals(existingID, passedID))
                 {
                     found = true;
                 }
+                else if (wi->GetAttribute()) {
+                    std::_tstring existingPath = wi->GetAttribute()->GetPath();
+                    if (boost::filesystem::exists(existingPath) && boost::algorithm::iequals(existingID, passedID))
+                    found = true;
```
Comment (GordonSmith) [inline eclide/ChildBuilderFrame.cpp:1166]: Indentation

File: wlib/EclCommand.h [hunk 5]
```diff
@@ -381,6 +384,27 @@ class CEclCommandMixin : public CMessageMap
         pT->PathToClipboard();
         return 0;
     }
+    LRESULT BookMarkInsertText(std::_tstring bookmarkStr) {
```
Comment (GordonSmith) [inline wlib/EclCommand.h:387]: Change to `const std::_tstring & bookmarkStr`

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: eclide/BookmarksView.cpp (hunk 3 above)
Comment (dehilsterlexis) [reply eclide/BookmarksView.cpp:516]: This is correct. This is checking to see if the description in the bookmark is different. If so, it updates the display.

**Review by GordonSmith — APPROVED** [review]

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**